Add make target for the Cray compilers

mgduda opened the pull request:
Running `make cray` will build SMIOL with the ftn and cc compilers.

MiCurry commented:
I'm not sure I have access to a Cray compiler at this time, so if things run for you I can approve and rebase this pull request.

mgduda commented:
I'm able to use the new `cray` target to build on Blue Waters, but because SMIOL calls MPI routines and the project allocation I was under on Blue Waters has expired, I can't actually run. Being able to at least compile is good enough that I think this PR is worth rebasing and merging now.

mgduda commented:
@MiCurry I updated the list of compiler names shown if no compiler target is specified, and I rebased the branch, so I think we're all set to do a fast-forward merge.
